### ISSUE TITLE
visual-mode command fails on master

### ISSUE THREAD

**Issue by Gavinok** [atlas-engineer/nyxt#2284]
**Describe the bug**

As of version 3-962-ga820793e the user can not properly enter visual
mode. A cursor is still placed on screen but no hints are shown.

The error message
```
 WARN - Warning: Error on separate thread: There is no applicable method for the generic function
  #&lt;STANDARD-GENERIC-FUNCTION NYXT/HINT-MODE:HINTS-SELECTOR (2)&gt;
when called with arguments
  (NIL).
See also:
  The ANSI Standard, Section 7.6.6
```
is instead printed

The error seems to come from the line

https://github.com/atlas-engineer/nyxt/blob/4a5db94e03444c84b9295fa1117dcdff888d53ab/source/mode/visual.lisp#L143

Since `(find-submode 'visual-mode)` returns `nil`

NOTE: This likely has to do with the recent changes to hint-mode and prompter. 

**Precise recipe to reproduce the issue**

1. Open Nyxt
2. Hit C-space
3. Type visual-mode
4. Hit enter

For website-specific issues:
Can you reproduce this issue with Epiphany / GNOME Web (https://wiki.gnome.org/Apps/Web)?

**Information**
- OS name+version: EndeavourOS
- Graphics card and driver: Intel WhiskeyLake-U GT2
- Desktop environment / Window manager name+version: Stumpwm 20.11-131-g0cc4d4f Compiled On Thu May 12 2022 15:12:51
- How you installed Nyxt (Guix pack, package manager, build from source): AUR
- Information from `show-system-information`:
Nyxt version: 3-962-ga820793e
Renderer version: GI-GTK
Operating system kernel: Linux 5.15.41-1-lts
Lisp implementation: SBCL 2.2.4 (Dynamic space size: 3221225472)
Features: (:WEBKIT2 :WEBKIT2-2.36 :WEBKIT2-CORS-ALLOWLIST :WEBKIT2-PASTE-PLAINTEXT :WEBKIT2-TRACKING :WEBKIT2-MUTE :WEBKIT2-EMOJI :WEBKIT2-MEDIA :WEBKIT2-SANDBOXING :GTK-3-22 :GTK-3-20 :GTK-3-18 :GTK-3-16 :GTK-3-14 :GTK-3-12 :GTK-3-10 :GTK-3-8 :GTK-3-6 :GTK-3-4 :GTK :GDK-3-22 :GDK-3-20 :GDK-3-18 :GDK-3-16 :GDK-3-14 :GDK-3-12 :GDK-3-10 :GDK-3-8 :GDK-3-6 :GDK-3-4 :CAIRO-1-10 :CAIRO-1-12 :GDK-PIXBUF :GLIB-2-30 :GLIB-2-32 :GLIB-2-34 :GLIB-2-36 :GLIB-2-38 :GLIB-2-40 :GLIB-2-42 :GLIB-2-44 :GLIB-2-46 :GLIB-2-48 :GLIB-2-50 :GLIB-2-52 :GLIB-2-54 :GLIB-2-56 :GLIB-2-58 :GLIB :NYXT-UNSTABLE :NYXT-GA820793E :NYXT-3 :FSET-EXT-STRINGS :CUSTOM-HASH-TABLE-NATIVE :SWANK :PLUMP-UTF-32 :PARENSCRIPT :GLOBAL-VARS :DECLARE-TYPES :SBCL+SAFE-STANDARD-READTABLE :NAMED-READTABLES :LPARALLEL :21BIT-CHARS :CLOSER-MOP :CL-PPCRE-UNICODE :CL-UNICODE :CL-JSON-DOUBLE-FLOAT-IS-SUBSUMED :CL-JSON-SINGLE-FLOAT-IS-SUBSUMED :CL-PPCRE :CHUNGA :FLEXI-STREAMS :BORDEAUX-THREADS :LPARALLEL.WITH-CLTL2 :LPARALLEL.WITH-CAS :LPARALLEL.WITH-STEALING-SCHEDULER :SPLIT-SEQUENCE CFFI-FEATURES:FLAT-NAMESPACE CFFI-FEATURES:X86-64 CFFI-FEATURES:UNIX :CFFI CFFI-SYS::FLAT-NAMESPACE ALEXANDRIA::SEQUENCE-EMPTYP :FAST-IO-SV :FAST-IO :CL-JSON-CLOS :CL-JSON :SBCL-USES-SB-ROTATE-BYTE CHIPZ-SYSTEM:GRAY-STREAMS :THREAD-SUPPORT :ASDF3.3 :ASDF3.2 :ASDF3.1 :ASDF3 :ASDF2 :ASDF :OS-UNIX :NON-BASE-CHARS-EXIST-P :ASDF-UNICODE :X86-64 :GENCGC :64-BIT :ANSI-CL :COMMON-LISP :ELF :IEEE-FLOATING-POINT :LINUX :LITTLE-ENDIAN :PACKAGE-LOCAL-NICKNAMES :SB-CORE-COMPRESSION :SB-LDB :SB-PACKAGE-LOCKS :SB-THREAD :SB-UNICODE :SBCL :UNIX)

ASDF version: 3.3.1
ASDF registries: (NYXT-SOURCE-REGISTRY ENVIRONMENT-SOURCE-REGISTRY)
Critical dependencies: (/home/gavinok/.cache/yay/nyxt-browser-git/src/nyxt-browser/_build/cl-cffi-gtk/gtk/cl-cffi-gtk.asd /home/gavinok/.cache/yay/nyxt-browser-git/src/nyxt-browser/_build/cl-gobject-introspection/cl-gobject-introspection.asd /home/gavinok/.cache/yay/nyxt-browser-git/src/nyxt-browser/_build/cl-webkit/webkit2/cl-webkit2.asd)

**Output when started from a shell**
<INFO> [13:20:41] Listening to socket #P"".
Nyxt version 3-962-ga820793e
<INFO> [13:20:41] Loading Lisp file #P"/home/gavinok/.config/nyxt/auto-config.lisp".
<INFO> [13:20:41] GTK extensions directory: #P"/home/gavinok/.cache/yay/nyxt-browser-git/src/nyxt-browser/libraries/web-extensions/"
<INFO> [13:20:41] GTK extensions directory: #P"/home/gavinok/.cache/yay/nyxt-browser-git/src/nyxt-browser/libraries/web-extensions/"
<INFO> [13:20:41] Loading #P"/home/gavinok/.local/share/nyxt/history/default.lisp".
<WARN> [13:20:45] Warning: Error on separate thread: There is no applicable method for the generic function
  #&lt;STANDARD-GENERIC-FUNCTION NYXT/HINT-MODE:HINTS-SELECTOR (2)&gt;
when called with arguments
  (NIL).
See also:
  The ANSI Standard, Section 7.6.6

**Comment by Ambrevar** [comment]
`find-submode` now only accepts package-prefix symbols (full symbols).  Wanna send a patch?

**Comment by aadcg** [comment]
I don't think that's the issue @Ambrevar.

@Gavinok, please read I [comment](https://github.com/atlas-engineer/nyxt/issues/2267#issuecomment-1133422499) I did.

**Comment by Gavinok** [comment]
@aadcg ya that comment lines up with my findings yesterday.

**Comment by Ambrevar** [comment]
Fixed with 92a7d7c0f75e4cfaa611f0f67316ff666eaeceb3.
